server/controller/src/main/resources/mapper/RuntimeVersionMapper.xml

```diff
@@ -78,39 +94,62 @@
     limit 1
   </select>
 
-  <insert id="revertTo">
+  <select id="findByVersionOrderAndRuntimeId" resultMap="runtimeVersionResultMap">
+    <if test="runtimeId != null">
+      and rv.runtime_id = #{runtimeId}
+    </if>
+    <include refid="select_runtime_version"/>
+    and rv.version_order = #{versionOrder}
+  </select>
+
+  <update id="revertTo">
     update runtime_version
-    set version_order = (select max from ((select max(version_order) as max from runtime_version))
-    as v) + 1
+    set version_order = if((select version_order
+                            from ((select version_order
+                                   from runtime_version
+                                   where id = #{rtVersionId})) as v1)
+                             = (select max
+                                from (select max(version_order) as max
+                                      from runtime_version
+                                      where runtime_id = (select runtime_id
+                                                          from runtime_version
```

Comment:
why not use `where runtime_id=#{rtId}` instread of ` where runtime_id = (select runtime_id
                                                          from runtime_version
                                                          where id = #{rtVersionId})) as v2)`